packages/input/src/Input/Input.tsx

```diff
@@ -0,0 +1,157 @@
+          onPointerDown={(event) => {
+            const target = event.target as HTMLElement;
+
+            // Make sure we're not clicking on an interactive element
+            if (target.closest("button, a")) {
+              event.preventDefault();
+              return;
+            }
+
+            const input = inputRef.current;
+            if (!input) return;
+
+            requestAnimationFrame(() => {
+              input.focus();
+            });
+          }}
+        >
+          <input
+            className={clsx(
+              "appearance-none text-gray-12 border-none shadow-0 outline-0 bg-transparent",
+              "[font-family:inherit] h-full w-full",
+              "order-2",
+              SIZE.Input[size],
+              className,
+            )}
+            disabled={disabled}
+            {...props}
+            ref={composedRefs}
+          />
+          {children}
+        </div>
+      </InputContext.Provider>
+    );
+  },
+);
+
+type SlotProps = React.ComponentPropsWithoutRef<typeof RadixSlot> & {
+  size?: keyof typeof SIZE.Slot;
+  position?: "leading" | "trailing";
+};
+type SlotRef = React.ElementRef<typeof RadixSlot>;
+
+const Slot = React.forwardRef<SlotRef, SlotProps>(
+  ({ position = "leading", ...props }, forwardedRef) => {
+    const context = React.useContext(InputContext);
+    return (
+      <div
+        className={clsx(
+          "group box-border flex items-center justify-center h-full",
+          "[&>[data-tgph-button]]:w-full [&>[data-tgph-button]]:h-auto",
+          // Overrides to match the icon button in figma
+          "[&>[data-tgph-button-layout='icon-only']]:aspect-square [&>[data-tgph-button-layout='icon-only']]:p-0",
+          // Overrides default button layout to match the button in figma
+          "[&>[data-tgph-button-layout='default']]:px-2",
+          // If only an icon button is present, set the aspect ratio to square
+          "[&:has([data-tgph-button-layout='icon-only'])]:aspect-square",
+          // If only an icon button is present, reset the padding to spacing-1
+          "[&:has([data-tgph-button-layout='icon-only'])]:!p-1",
+          position === "leading" && SIZE.SlotLeading[context.size],
+          position === "trailing" && SIZE.SlotTrailing[context.size],
+          SIZE.Slot[props.size ?? context.size],
+        )}
+      >
+        <RadixSlot size={context.size} {...props} ref={forwardedRef} />
+      </div>
+    );
+  },
+);
+
+type DefaultProps = React.ComponentPropsWithoutRef<typeof Root> & {
+  LeadingComponent?: React.ReactNode;
+  TrailingComponent?: React.ReactNode;
+};
+
+const Default = ({
+  LeadingComponent,
+  TrailingComponent,
+  ...props
+}: DefaultProps) => {
+  return (
+    <Root {...props}>
+      {LeadingComponent && <Slot position="leading">{LeadingComponent}</Slot>}
+      {TrailingComponent && (
+        <Slot position="trailing">{TrailingComponent}</Slot>
+      )}
+    </Root>
+  );
+};
```

Comment:
This component is only for making composition of the input quicker like this:
```
<Input size="3" LeadingComponent={<Button} />
```

You can always drop down to the individual parts and build an input like this:

Instead of utilizing the underlying components like this
```
<Input.Root>
    <Input.Slot position="leading">
        <Button/>
     </Input.Slot>
</Input.Root>